extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -194,9 +199,11 @@ private static QuarkusSecurityIdentity augmentIdentity(SecurityIdentity security
 
         // Client secret has to be posted as a form parameter if OIDC requires the client_secret_post authentication
         Credentials creds = configContext.oidcConfig.getCredentials();
-        if (creds.clientSecret.value.isPresent() && creds.clientSecret.method.isPresent()
-                && Secret.Method.POST == creds.clientSecret.method.get()) {
+        if (creds.clientSecret.value.isPresent() && Secret.Method.POST == creds.clientSecret.method.orElse(null)) {
             params.put("client_secret", creds.clientSecret.value.get());
+        } else if (creds.jwt.secret.isPresent()) {
+            params.put("client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer");
+            params.put("client_assertion", signJwtWithClientSecret(configContext.oidcConfig));
```

Comment:
I'm wondering if we can't just create the signed JWT at build time? But I guess that would not work for dynamic configurations?

Reply:
@pedroigor Good idea, I can calculate it in `OidcRecorder` and save it in `TenantConfigContext`, if the secret is updated in the dev mode then it will be recalculated